core/file_walker.go

```diff
@@ -4,17 +4,57 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 )
 
 type File struct {
-	Name  string
-	Size  int64
-	IsDir bool
-	Files []*File
+	Name     string
+	Size     int64
+	IsDir    bool
+	Files    []*File
+	IsMarked bool
 }
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
+func (f *File) ToggleIsMarked() bool {
+	f.IsMarked = !f.IsMarked
+	return f.IsMarked
+}
+
+func GetMarked(root File) []string {
+	var marked []string
+	var wg sync.WaitGroup
+	c := make(chan string, 1)
+	wg.Add(1)
+	go func() {
+		wg.Wait()
+		close(c)
+	}()
+	go getMarkedRecursive("", &root, c, &wg)
```

Comment:
I meant unit tests. When I run `core.StartProcessing` as a separate go routine it made the testing much harder (https://github.com/viktomas/godu/blob/master/core/processor_test.go) Certain test cases don't even expect anything and they succeed when no deadlock happens (which is not optimal IMHO).

But that might as well be just me being new to golang. You might find it as easy to unit test go routines as normal functions..